tools/c7n_index/c7n_index/metrics.py

```diff
@@ -229,6 +278,55 @@ def index_account(config, idx_name, region, account, start, end, period):
     return region_time, region_points
 
 
+def index_account_resources(config, account, region, policy, date):
```

Comment:
a good portion of the logic here feels duplicated from c7n/report/csvoutput.py record_set func.. i'd rather use that implementation rather than encoding key structure assumptions multiple times. if there's deltas on why that won't work lets try to address that in record_set